### PR TITLE
Add support for .deb and .rpm files

### PR DESCRIPTION
**Pull request by egibs**
Closes: #667 

This PR adds extraction support for `.deb` and `.rpm` files. The new packages used are essentially doing the same things we're doing for the other extraction methods.

I also updated `extractTar` to use the `Header.Typeflag` values since it's cleaner than checking for the header's file mode.

Examples:
`.deb`:
```
$go run cmd/mal/mal.go analyze ~/Downloads/teleport_16.4.8_amd64.deb
🔎 Scanning "~/Downloads/teleport_16.4.8_amd64.deb"
├─ 😈 ~/Downloads/teleport_16.4.8_amd64.deb ∴ /usr/local/bin/tsh [CRITICAL]
│     ≡ collection [MEDIUM]
│       🟡 archives/zip — Works with zip files: archive/zip
│       🟡 databases/mysql — accesses MySQL databases: mysql
│       🟡 databases/postgresql — accesses PostgreSQL databases: postgresql
│       🔵 databases/sql — accesses SQL databases
│     ≡ command & control [MEDIUM]
│       🟡 addr/discord — may report back to 'Discord'
│       🟡 addr/http_dynamic — URL that is dynamically generated:
│             http://%s, https://%s/web/cluster/, https://%s/webapi/findhttps, https://%s/webapi/motdunexpected,…
│       🟡 addr/ip — mentions an IP and port:
│             browserIp, browser_ip, cIp, client_ip, dst_port, eIp, endPort, existsPort, fromPort, from_port, hI…
│       🟡 addr/server — references a 'server address', possible C2 client:
│             preconditionsserverAddressincludeObjectfieldSelectorman, serverAddressByClientCIDRsdeletionGracePe…
...
```
`.rpm`:
```
$go run cmd/mal/mal.go analyze ~/Downloads/teleport-17.0.2-1.x86_64.rpm 
🔎 Scanning "~/Downloads/teleport-17.0.2-1.x86_64.rpm"
├─ 🟡 ~/Downloads/teleport-17.0.2-1.x86_64.rpm ∴ /usr/local/bin/tctl [MEDIUM]
│     ≡ collection [MEDIUM]
│       🟡 archives/unarchive — unarchives files: unarchive_unix
│       🟡 archives/zip — Works with zip files: archive/zip
│       🟡 databases/mysql — accesses MySQL databases: mysql
│       🟡 databases/postgresql — accesses PostgreSQL databases: postgresql
│       🔵 databases/sql — accesses SQL databases
│       🟡 databases/sqlite — accesses SQLite databases: sqlite3
│     ≡ command & control [MEDIUM]
...
```

`.tar.gz` (verifying functionality after change):
```
$ go run cmd/mal/mal.go analyze ~/Downloads/teleport-access-slack-v17.0.2-linux-amd64-bin.tar.gz
🔎 Scanning "~/Downloads/teleport-access-slack-v17.0.2-linux-amd64-bin.tar.gz"
├─ 🟡 ~/Downloads/teleport-access-slack-v17.0.2-linux-amd64-bin.tar.gz ∴ /teleport-access-slack/install [MEDIUM]
│     ≡ execution [MEDIUM]
│       🔵 plugin — references a 'plugin': Plugins, plugins
│       🟡 shell/exec — executes shell: /bin/sh
│     ≡ filesystem [MEDIUM]
│       🔵 directory/create — creates directories: mkdir
│       🟡 file/copy — forcibly copy files using cp -f: cp -f teleport-slack $BINDIR/
│       🔵 path/etc — path reference within /etc: /etc/teleport-slack.toml
│       🟡 path/usr_local — path reference within /usr/local/bin
│       🔵 path/var — path reference within /var: /var/lib/teleport/plugins/slack
│     ≡ process [LOW]
│       🔵 chdir — changes working directory: cd $
│
├─ 🟡 ~/Downloads/teleport-access-slack-v17.0.2-linux-amd64-bin.tar.gz ∴ /teleport-access-slack/teleport-slack [MEDIUM]
│     ≡ collection [MEDIUM]
│       🟡 archives/zip — Works with zip files: archive/zip
│       🟡 databases/mysql — accesses MySQL databases: mysql
│       🔵 databases/sql — accesses SQL databases
│     ≡ command & control [MEDIUM]
│       🟡 addr/discord — may report back to 'Discord'
│       🟡 addr/http_dynamic — URL that is dynamically generated: https://%s/webapi/findunexpected
│       🟡 addr/ip — mentions an IP and port:
│             browserIp, browser_ip, client_ip, dst_port, fromPort, from_port, hasPort, ipPort, ip_port, lIp, lo…
│       🟡 addr/server — references a 'server address', possible C2 client: serverUrl, server_addr
...
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,6 +20,7 @@ require (
 	golang.org/x/sync v0.9.0
 	golang.org/x/term v0.26.0
 	gopkg.in/yaml.v3 v3.0.1
+	pault.ag/go/debian v0.17.0
```
Comment (egibs) [inline go.mod:23]: There aren't really any `.deb` packages that are current; most are 7+ years old.

File: go.mod [hunk 2]
```diff
@@ -30,6 +33,7 @@ require (
 	github.com/docker/docker-credential-helpers v0.8.2 // indirect
 	github.com/ebitengine/purego v0.8.1 // indirect
 	github.com/go-ole/go-ole v1.3.0 // indirect
+	github.com/kjk/lzma v0.0.0-20161016003348-3fd93898850d // indirect
```
Comment (egibs) [inline go.mod:36]: > 2016
:(

File: go.mod [hunk 3]
```diff
@@ -48,8 +52,11 @@ require (
 	github.com/tklauser/go-sysconf v0.3.14 // indirect
 	github.com/tklauser/numcpus v0.9.0 // indirect
 	github.com/vbatts/tar-split v0.11.6 // indirect
+	github.com/xi2/xz v0.0.0-20171230120015-48954b6210f8 // indirect
```
Comment (egibs) [inline go.mod:55]: See above.

File: go.mod [hunk 4]
```diff
@@ -7,6 +7,7 @@ require (
 	github.com/cavaliergopher/cpio v1.0.1
 	github.com/cavaliergopher/rpm v1.2.0
 	github.com/chainguard-dev/clog v1.5.0
+	github.com/egibs/go-debian v0.18.0
```
Comment (egibs) [inline go.mod:10]: I ended up forking the upstream package and updating the dependencies.

Comment (egibs) [reply go.mod:10]: Functionality is unchanged:
```
go run cmd/mal/mal.go scan ~/Downloads/teleport_16.4.8_amd64.deb
🔎 Scanning "~/Downloads/teleport_16.4.8_amd64.deb"
├─ 😈 ~/Downloads/teleport_16.4.8_amd64.deb ∴ /usr/local/bin/tsh
│     • crypto/blockchain — blockchain&{0x140000ae120}
│     • impact/remote_access/reverse_shell — reverse shell in Perl: ">&, open(, sh -i, socket(
├─ 🛑 ~/Downloads/teleport_16.4.8_amd64.deb ∴ /usr/local/bin/fdpass-teleport
│     • sus/compiler — built with multiple versions of GCC: GCC: (GNU) 12.1.1 20220628 (Red Hat 12.1.1-3), GCC: (GNU) 4.8.5 20150623 (Red Hat 4.8.5-44)
├─ 🛑 ~/Downloads/teleport_16.4.8_amd64.deb ∴ /usr/local/bin/teleport
│     • c2/refs — Uses terms that may reference a command and control server: dinvalid ec2 address, is requiredec2 address, rkInterfaceec2 address
│     • credential/ssh/d_memory_map — May access the memory map for sshd: passwd, password, ptrace, tracer
│     • crypto/blockchain — blockchain&{0x140000ae120}
│     • evasion/process_injection/ptrace — known ptrace injectors: BPF, PERF_SAMPLE
│     • exfil/stealer/creds — suspected data stealer: Atomic, Bookmarks, Chrome, Chromium, Discord, Firefox, History, Snowflake

💡 For detailed analysis, try "mal analyze <path>"
```

**Review by tstromberg — APPROVED** [review]

**Pull request merged**